data-prepper-plugins/aggregate-processor/src/test/java/com/amazon/dataprepper/plugins/processor/aggregate/AggregateProcessorIT.java

```diff
@@ -0,0 +1,191 @@
+
+    @Mock
+    private PluginFactory pluginFactory;
+
+    @BeforeEach
+    void setup() {
+        aggregatedResult = new ConcurrentLinkedDeque<>();
+        uniqueEventMaps = new HashSet<>();
+
+        final List<String> identificationKeys = new ArrayList<>();
+        identificationKeys.add("firstRandomNumber");
+        identificationKeys.add("secondRandomNumber");
+        identificationKeys.add("thirdRandomNumber");
+
+        aggregateConfigMap = new HashMap<>();
+        aggregateConfigMap.put("identification_keys", identificationKeys);
+        aggregateConfigMap.put("action", new PluginModel(UUID.randomUUID().toString(), Collections.emptyMap()));
+
+        aggregateAction = new RemoveDuplicatesAggregateAction();
+
+        eventBatch = getBatchOfEvents();
+
+        pluginMetrics = PluginMetrics.fromNames(UUID.randomUUID().toString(), UUID.randomUUID().toString());
+
+        when(pluginFactory.loadPlugin(eq(AggregateAction.class), any(PluginSetting.class)))
+                .thenReturn(aggregateAction);
+    }
+
+    private AggregateProcessor createObjectUnderTest() {
+        final AggregateProcessorConfig aggregateProcessorConfig = OBJECT_MAPPER.convertValue(aggregateConfigMap, AggregateProcessorConfig.class);
+        return new AggregateProcessor(aggregateProcessorConfig, pluginMetrics, pluginFactory);
+    }
+
+    @RepeatedTest(value = 10)
+    void aggregateWithNoConcludingGroupsReturnsExpectedResult() throws InterruptedException {
+        final AggregateProcessor objectUnderTest = createObjectUnderTest();
+
+        final ExecutorService executorService = Executors.newFixedThreadPool(NUM_THREADS);
+        final CountDownLatch countDownLatch = new CountDownLatch(NUM_THREADS);
+
+        for (int i = 0; i < NUM_THREADS; i++) {
+            executorService.execute(() -> {
+                final List<Record<Event>> recordsOut = (List<Record<Event>>) objectUnderTest.doExecute(eventBatch);
+                for (final Record<Event> record : recordsOut) {
+                    final Map<String, Object> map = record.getData().toMap();
+                    aggregatedResult.add(map);
+                }
+                countDownLatch.countDown();
+            });
+        }
+
+        boolean allThreadsFinished = countDownLatch.await(5L, TimeUnit.SECONDS);
+
+        assertThat(allThreadsFinished, equalTo(true));
+        assertThat(aggregatedResult.size(), equalTo(NUM_UNIQUE_EVENTS_PER_BATCH));
+
+        for (final Map<String, Object> uniqueEventMap : uniqueEventMaps) {
+            assertThat(aggregatedResult, hasItem(uniqueEventMap));
+        }
+
+        for (final Map<String, Object> eventMap : aggregatedResult) {
+            assertThat(eventMap, in(uniqueEventMaps));
+        }
+    }
+
+    @RepeatedTest(value = 2)
+    void aggregateWithConcludingGroupsOnceReturnsExpectedResult() throws InterruptedException {
+        final AggregateAction spyAction = spy(aggregateAction);
```

Comment:
What does this spy do?

Reply:
Thanks for catching that. I was experimenting with it and forgot to remove it.